gluten-core/src/main/scala/org/apache/spark/shuffle/ShuffleManagerRouter.scala

```diff
@@ -0,0 +1,137 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.shuffle
+import org.apache.spark.{ShuffleDependency, TaskContext}
+import org.apache.spark.network.buffer.ManagedBuffer
+import org.apache.spark.network.shuffle.MergedBlockMeta
+import org.apache.spark.storage.{BlockId, ShuffleBlockBatchId, ShuffleBlockId, ShuffleMergedBlockId}
+
+/** The internal shuffle manager instance used by GlutenShuffleManager. */
+private class ShuffleManagerRouter(lookup: ShuffleManagerLookup) extends ShuffleManager {
+  import ShuffleManagerRouter._
+  private val cache = new Cache()
```

Comment:
Will this status be sensed by spark executor?

Reply:
Will the mapping saved in `cache.store` also be synchronized to the executor? I am worried that calling `cache.get(shuffleId)` on the executor will not be able to find the ShuffleManager

Reply:
I see. It's likely a problem. Let me test, thanks.

Reply:
Perhaps a simple solution could be register `ShuffleHandle -> ShuffleManager` mappings as well which can be used for lookup on executor side.. Would you think that is feasible?

Reply:
@wForget I assume the current code is fine, given that usually shuffle dependency is carried by shuffle RDD and called from task code. Will run some tests later to prove.

https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/rdd/ShuffledRDD.scala

Reply:
> Perhaps a simple solution could be registering `ShuffleHandle -> ShuffleManager` mappings as well which can be used for lookup on executor side. Would you think that is feasible?

Sounds feasible